internal/quaiapi/api.go

```diff
@@ -1263,6 +1264,27 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 		to = *args.To
 	} else {
 		to = crypto.CreateAddress(args.from(nodeLocation), uint64(*args.Nonce), *args.Data, nodeLocation)
+		if _, err := to.InternalAndQuaiAddress(); err != nil {
```

Comment:
Why are we implementing address grinding in the `AccessList()` method? Address grinding should only happen in the create methods and their helpers.

Reply:
We have to determine what the 'to' address is to add it to the tracer

Reply:
just move the logic to a helper, as mentioned in your other PR. Repeating code like this in two places is how we end up with differences between the two places

Reply:
Fixed